util/buildinfo/export.go

```diff
@@ -0,0 +1,44 @@
+package buildinfo
+
+import "github.com/pkg/errors"
+
+// ExportMode represents the export mode for buildinfo opt.
+type ExportMode int
+
+const (
+	// ExportNone doesn't export build dependencies.
+	ExportNone ExportMode = 1 << iota
+	// ExportImageConfig exports build dependencies to
+	// the image config.
+	ExportImageConfig
+	// ExportMetadata exports build dependencies as metadata to
+	// the exporter response.
+	ExportMetadata
+	// ExportAll exports build dependencies as metadata and
+	// image config.
+	ExportAll = -1
+)
+
+// ExportDefault is the default export mode for buildinfo opt.
+const ExportDefault = ExportAll
+
+// ParseExportMode returns the export mode matching a string.
+func ParseExportMode(s string) (ExportMode, error) {
+	switch s {
+	case "none":
+		return ExportNone, nil
+	case "imageconfig":
+		return ExportImageConfig, nil
+	case "metadata":
+		return ExportMetadata, nil
+	case "all":
+		return ExportAll, nil
+	default:
+		return 0, errors.Errorf("unknown buildinfo export mode: %s", s)
+	}
+}
+
+// Can checks if current mode has export capability
+func (e ExportMode) Can(tgt ExportMode) bool {
```

Comment:
Don't think this is needed. More clear to just do the bitwise and inline. `&ExportAll` is not needed as well as it is defined as `-1` so has no limiting capability anyway.

Reply:
Yeah not needed, just wanted an helper for this logic.